### PR TITLE
Replaced the name with the pipeline_step_name within the repr of a St…

### PR DESCRIPTION
**Pull request by AlexejPenner**
…epView

## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [x] If my change requires a change to the documentation, I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Describe changes
Fixing the problem of the post execution work
flow where the name displayed within the step view is not the name that can be used with the .get_step(name=...) method of a pipeline run.

stepnames

### CODE REVIEW

**Review by htahir1 — COMMENTED** [review]
I actually now did a deep dive into how we use `name` and `pipeline_run_name` and maybe we need to come up to a definition that is clearer. My proposal would be that we replace `pipeline_step_name` to `name` and what we refer to as `name` here should be something like `step_functional_name`. That way we make it's clearer I think when you fetch step by name in the `PipelineRunView`.

**Comment by schustmi** [comment]
When we tried to come up with names for these two we had similar issues, but the problem with a name like `functional_name` or `step_function_name` is that it disregards the class-based API (and similarly `class_name` etc.)

**Comment by htahir1** [comment]
@schustmi  How about we then call it `entrypoint_name`?  I still feel like `name` should be by default the name within the pipeline itself rather than the decorated function

**Comment by schustmi** [comment]
@htahir1 Agreed on the name swapping (I can remember that we tried to have it that way but simply couldn't come up with fitting names if we swapped them). `entrypoint_name` sounds reasonable but to me at least it wouldn't be intuitively clear what `name` and `entrypoint_name` refer to when looking at the `StepView` class.

**Comment by htahir1** [comment]
@schustmi I mean most users will mostly not be using `entrypoint_name` anyway right? they would be using `name` and if we make it clear that `name` is the thing defined in the pipeline, than they will mostly just use that, and see the entrypoint_name if they are just curious. At that point, it should be clear anyway/

P.S. @AlexejPenner I would add `entrypoint_name` also to the `__repr__` to make this very clear

**Comment by schustmi** [comment]
@htahir1 Makes sense, let's go with that

File: examples/quickstart/quickstart.py [hunk 1]
```diff
@@ -33,7 +33,7 @@ def importer() -> Output(
 
 
 @step
-def trainer(
```
Comment (schustmi) [inline examples/quickstart/quickstart.py:36]: Is this a typo or an intended change? :D

File: src/zenml/cli/example.py [hunk 2]
```diff
@@ -226,7 +226,7 @@ def latest_release(self) -> str:
         latest_tag = parse(tags[-1].name)
         if type(latest_tag) is not Version:
             return "main"
-        return tags[-1].name  # type: ignore
```
Comment (schustmi) [inline src/zenml/cli/example.py:229]: I think here it should have stayed `name`.

File: src/zenml/integrations/kubeflow/container_entrypoint.py [hunk 3]
```diff
@@ -203,7 +203,7 @@ def _dump_input_populated_artifacts(
             )
             # There must be at least a channel in a input, and all channels in
             # a input share the same artifact type.
-            artifact_type = spec.channels[0].artifact_query.type.name  # type: ignore[attr-defined] # noqa
+            artifact_type = spec.channels[0].artifact_query.type.entrypoint_name  # type: ignore[attr-defined] # noqa
```
Comment (schustmi) [inline src/zenml/integrations/kubeflow/container_entrypoint.py:206]: I think all three changes in this kubeflow entrypoint don't use the post-execution step class and therefore should stay `name`.

**Review by schustmi — CHANGES_REQUESTED** [review]

**Comment by AlexejPenner** [comment]
@schustmi @htahir1 i have implemented your requested naming, this went through multiple places in the code - especially in the visualizers the entrypoint_name seems to have been used. I verified all affected examples run on my machine.

File: src/zenml/integrations/kubeflow/container_entrypoint.py (hunk 3 above)
Comment (AlexejPenner) [reply src/zenml/integrations/kubeflow/container_entrypoint.py]: removed

File: src/zenml/cli/example.py (hunk 2 above)
Comment (AlexejPenner) [reply src/zenml/cli/example.py]: fixed this

File: examples/quickstart/quickstart.py (hunk 1 above)
Comment (AlexejPenner) [reply examples/quickstart/quickstart.py]: fixed this

**Review by schustmi — APPROVED** [review]
LGTM

**Review by htahir1 — APPROVED** [review]
This LGTM! Hopefully nothing breaks! :-D

**Pull request merged**